redaxo/src/core/lib/console/application.php

```diff
@@ -26,4 +27,26 @@ public function doRun(InputInterface $input, OutputInterface $output)
             throw new FatalThrowableError($e);
         }
     }
+
+    protected function doRunCommand(Command $command, InputInterface $input, OutputInterface $output)
+    {
+        if ($command instanceof rex_console_command) {
+            $this->loadPackages($command);
+        }
+
+        return parent::doRunCommand($command, $input, $output);
+    }
+
+    private function loadPackages(rex_console_command $command)
+    {
+        if ($command->requiresOtherPackages()) {
+            require rex_path::core('packages.php');
```

Comment:
spätestens an diesem punkt haben wir das gleiche problem wie ohne diesen PR, oder?

Reply:
Oder können wir im deploy case jedes addon ohne abh. Laden?

Reply:
Der migrate-Command wird dies enthalten:

```php
public function requiresOtherPackages()
{
    return false;
}
```

Reply:
Anders gesagt: diesen neuen modus brauchen wir nur in ydeploy, sonst für kein anderes addon?

Reply:
Ja genau. Zumindest fällt mir zurzeit kein anderer Fall ein. Auch hier in der Liste sehe ich keinen Fall: https://github.com/redaxo/redaxo/issues/1114
(Außer theoretisch Setup, wobei im Setup-Modus sowieso schon nur backup und be_style geladen werden.)

Wir können auch gerne nochmal überlegen, ob uns ein Weg einfällt, wie es ydeploy lösen kann, ohne dass der Core dafür was vorsieht. Bisher fiel mir aber nichts so wirklich tolles ein.

Reply:
Also ich werde das hier wohl so mergen, denn ich sehe nicht wirklich ne Alternative..

Reply:
Wenn es nur für das deploy im speziellen ist würde ich ggf, weniger allg. Namen verwenden

Reply:
Wie meinst du das?

Reply:
Der mechanismus ist aktuell rel. Allg. - wir brauchen wir ihn aber nur für den sonderfall ydeploy. Ggf wird es einfacher wenn wir speziellere namen nehmen

Reply:
Hmm, kann mir gerade ehrlich gesagt nicht so richtig vorstellen, in welche Richtung du das meinst. Also wie konkret speziellere Namen hier aussehen könnten.
Finde der Name trifft es eigentlich gut.

Reply:
Z.b. Statt requiresOtherPackages dann isDeployCommand

Reply:
Hmm, glaube lasse es lieber so. Es ist übrigens noch verzwickter, `ydeploy:diff` braucht gerade auch die anderen Addons. Denn wir ändern zum beispiel teilweise im project-Addon die ydeploy-Config, die beim diff ne Rolle spiel.
Nur `ydeploy:migrate` muss ohne die Addons laufen.

Also du würdest quasi einfach vom Core aus festgebrannt schauen, wenn Command == `ydeploy:migrate`, dann nur ydeploy laden, ansonsten alle addons?

Reply:
Solange wir unsicher  sind wuerde ich wenig api code einbauen und lieber hard codieren, ja

Reply:
Ok, überzeugt. Und hast vor allem recht, Api später ergänzen, ist kein problem, Api wieder entfernen hingegen schon.

Reply:
Das ist der Punkt! 👍